p4include/v1model.p4

```diff
@@ -57,11 +57,8 @@ struct standard_metadata_t {
     @alias("intrinsic_metadata.mcast_grp")     bit<16> mcast_grp;
     @alias("intrinsic_metadata.resubmit_flag") bit<1>  resubmit_flag;
     @alias("intrinsic_metadata.egress_rid")    bit<16> egress_rid;
-}
-
-extern Checksum16 {
```

Comment:
Rather than deleting Checksum16 (which existing P4_16 programs depend on), it should be marked as `@deprecated`

Reply:
But the compiler support for this block has been deleted.
It won't do much help to have programs that use it.

Reply:
Unless the compiler errors for all deprecated constructs...

Reply:
That's kind of the point -- you shouldn't remove support for it yet, just add support for the new way of dealing with things.

And yes, there should be a pass that checks for uses of `@deprecated` things and issues a warning about it -- the `@deprecated` could have a string literal arg with a message suggesting what it should be changed to.

Reply:
The fundamental problem is that the old way of doing things was really broken; if I could have fixed that implementation I would have.

I should indeed add a pass for handing deprecated. It would be easiest to make it really an error rather than a warning. 

It will be quite difficult to support both variants, because not only the checksum extern changes, but also the signature of the verifyChecksum control.

Reply:
It was only broken in the bmv2 backend and was working fine in other backends.

Reply:
There are lots of filed issues about it for bmv2.

Also, I suspect there were no torture tests; technically you could have any combination of ifs, assignments and method calls in the update/verify checksum blocks. I somehow doubt that all these were handled.

Reply:
On the other hand, the new methods should be easily converted to the old style code...
`update_checksum(c, list, field, HashAlgorithm.csum16)` is the same as
`if (c) field = ck.get(list);`

Reply:
@ChrisDodd : assuming we merge PR #940 and we tag the Checksum16 block as deprecated, would this solution work for you? The bmv2 back-end would not support it, but your custom back-end could.